chore: migrate alignment bar chart to ngxchart

sodabottlebaby opened the pull request:
# Description

Migrated one of the visualisation components (alignment bar chart) to ngx charts used in 9.x

To properly migrate this component I also included the migration of outcome-service which was my initial task #302. 

Fixes # (issue)

## Type of change

_Please delete options that are not relevant._

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] This change requires a documentation update

# How Has This Been Tested?

Login using student_1 and navigate to a unit with learning outcomes. Proceed to alignment tab and click the bar chart button, hovering over the chart for better visibility.

edit: In the original 9.x branch, clicking the button won't work and the chart won't be visible. The console in Chrome DevTools shows this following error:
![Screenshot 2025-05-16 at 5 33 39 pm](https://github.com/user-attachments/assets/125b2e58-717b-4cf1-9804-972b671d9807)

To fix this missing dependency injection, you can simply add $sce on line 6 of outcome-service.coffee and the chart should be visible as below.

before the migration:
<img width="963" alt="Screenshot 2025-05-11 at 6 00 27 pm" src="https://github.com/user-attachments/assets/4c853178-1ac8-4135-8554-c5db730c165a" />
<img width="958" alt="Screenshot 2025-05-11 at 6 00 31 pm" src="https://github.com/user-attachments/assets/a5ebe3f9-66d9-4837-b3b4-12d519beb2f7" />

after the migration:
<img width="1225" alt="Screenshot 2025-05-26 at 3 38 42 pm" src="https://github.com/user-attachments/assets/e8e88d99-8b1b-43ba-bc77-39db6b4cf0f5" />
<img width="1066" alt="Screenshot 2025-05-26 at 3 40 47 pm" src="https://github.com/user-attachments/assets/1c368668-6105-4e8f-a5c1-f8f5efc27e86" />

## Testing Checklist:

- [x] Tested in latest Chrome
- [x] Tested in latest Safari
- [x] Tested in latest Firefox

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have requested a review from @disururathnayake and @zara1504 on the Pull Request

sodabottlebaby committed:
chore: migrate alignment bar chart to ngxchart

disururathnayake commented:
Hi @sodabottlebaby,

Migration looks good but there are few errors you have to resolve. I will list down those errors for you

### **Errors**

- tool tip is not working when hovering charts
- Filtering from the legend menu is not working. 
<img width="374" alt="image" src="https://github.com/user-attachments/assets/2af9cdbd-acf3-40c6-9f43-6dff0f870411" />


- Even there is no Learning Outcome chart should shown like below. 
<img width="881" alt="image" src="https://github.com/user-attachments/assets/a96b0ff3-d4ec-4625-bb6a-d48a9b6f289b" />


- Chart is not updating real time when selecting a new rating and deleting a existing rating.
<img width="881" alt="image" src="https://github.com/user-attachments/assets/e461c305-0a0f-4ad5-bb1c-83a1f13aab12" />


Please resolve these issues and I'll do a review again and give you an update.

zara1504 commented:
Hey Su, i just pulled your work and your code looks good and when I click on the Outcome Alignment and Visulization its loads like the images provided above. However, I did notice that when I click the grade filters it wont work and throws this error in the console. 

<img width="711" alt="Screenshot 2025-05-17 at 11 24 08 PM" src="https://github.com/user-attachments/assets/aa66e84a-0ece-457b-9ea1-19e886f0c1aa" />

Another thing to add is the before and after images you have shown are slightly different with the grade legend being on the side instead of above the chart. (If it was supposed to be a different layout then don't worry about this but I couldn't find anything about changing the layout of the page.) 

Overall, good job with it and let me know how it goes fixing that error :)

sodabottlebaby committed:
chore: migrate alignment bar chart to ngxchart

- Tool tip is now workinng when hovering charts
- Fixed the filtering error from the legend
- Added noBarWhenZero = "false" to display grades with values of zero
- Chart is now updating real time

sodabottlebaby commented:
> Hey Su, i just pulled your work and your code looks good and when I click on the Outcome Alignment and Visulization its loads like the images provided above. However, I did notice that when I click the grade filters it wont work and throws this error in the console.
> 
> <img alt="Screenshot 2025-05-17 at 11 24 08 PM" width="711" src="https://private-user-images.githubusercontent.com/139195494/444782978-aa66e84a-0ece-457b-9ea1-19e886f0c1aa.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDgyMzgzODQsIm5iZiI6MTc0ODIzODA4NCwicGF0aCI6Ii8xMzkxOTU0OTQvNDQ0NzgyOTc4LWFhNjZlODRhLTBlY2UtNDU3Yi05ZWExLTE5ZTg4NmYwYzFhYS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNTI2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDUyNlQwNTQxMjRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mNzcyNmZlMGNmYzAwZGEyOTVkOTlhYzViZDE2ZDU1OGMzM2M2NjA1MDhhN2IwOTIyZmE0OGUxZTkzNTA3YzQ1JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.Yk1DE0Z4RU2ODZlr61qOCWD28GCE4R4UjyX1U8-H-tE">
> Another thing to add is the before and after images you have shown are slightly different with the grade legend being on the side instead of above the chart. (If it was supposed to be a different layout then don't worry about this but I couldn't find anything about changing the layout of the page.)
> 
> Overall, good job with it and let me know how it goes fixing that error :)

I believe the layout difference is because of the ngxcharts, I did make some addition to the scss file but only for better viewing and the display of grouped and stacked buttons.

returnMarcco commented:
Marking this as stale. Will clear this PR off the MSPlanner ticket. Recommending this PR to be closed.
